js/main.js

```diff
@@ -646,31 +636,31 @@ function initiate() {
 
         /* STAGE: bar used instead of \mid in set comprehension */
         for (var i = 0; i < mathFragments.length; ++i) {
-            var badMatch = mathFragments[i].match(/\\\{.*?\|.*?\\}/);
-            if (badMatch && !badMatch[0].includes('\\mid')){
-                addWarning('MID_IN_SET_COMPREHENSION', null, badMatch[0]);
+            var errorRegex = /\\\{.*?\|.*?\\}/g;
```

Comment:
Same answer as before -- without the global flag the `exec` method would return the same thing over and over.